### ISSUE TITLE
add system dynamic variables $envIf

### ISSUE THREAD

**Issue by readless** [Huachao/vscode-restclient#319]
can you add an system dynamic variables $envIf, 
when i use @name= {{$envIf local name1  name2}}, it mean if current env is local , then name = name1 , otherwise name =name2。

this feture will be usable。

**Comment by Huachao** [comment]
@readless this seems can be achieved via current offerings.

1. Create one environment named `local` in vscode settings:
```json
"rest-client.environmentVariables": {
        "$shared": {
            "name": "readless"
        },
        "local": {
            "name": "Huachao"
        }
    }
```
This piece of code means, if your environment is local, then the value of `name` would be `Huachao`, otherwise, the value of name would be `readless`

2. Create a sample request like following:
```http
@userName = {{name}}
GET https://api.github.com/users/{{userName}} HTTP/1.1
``` 
3. Send this request in `local` environment and other environments, the response would be different

**Comment by readless** [comment]
if i hava two http file , 
A.http use two variables {{name}} {{code}} ,
when env = local, name=name1, code=code1
         env = dev, name=name2, code=code2
B.http use two variables {{name}} {{code}} 
when env = local, name=name3, code=code1
         env = dev, name=name4, code=code2

so code can be put in rest-client.environmentVariables, but name depends different http request。
if i  also  put name in  rest-client.environmentVariables, then i have four env。

**Comment by readless** [comment]
when i have 30 or more http files , i must have 30*2 = 60 env。  

i  will make a pull request

**Comment by Huachao** [comment]
@readless I agree with the idea for the conditional file variables whose value determined by some conditions. For your case, the condition is the assertion of environment name, I am considering if it would be better to provide a more general way to support conditional file variables. Just like the `@variableName = {{condition ? value1 : value2}}`. The condition can be the validation of current active environment, other responses even many other supported conditions. So only adding the `$envIf` variable seems not so general and useful to many users. I'd like to leave this issue and your great PR active, and let other users judge this. So before that, maybe you have to use your self-packaged extension to experience this feature. Sorry for the inconvenience.

**Comment by Meir017** [comment]
maybe rename the variable to `$if` and create condition keywords.
for example:
```
{{ $if env=local ? value1 : value2 }}
```
or maybe
```
{{ $if env local value1 value2 }}
```

**Comment by Huachao** [comment]
@Meir017 nice suggestion, @readless  actually I think the scenario is not quite common for many users, if we want to support what @Meir017 suggested, we'd better first create a new environment name called something similar to `$currentEnvironmentName`, then use this variable to do some validation.

**Comment by xdmushui** [comment]
Maybe put variables to file.
For example:

```
// file: variables.http 

@base_url = https://localhost:44366
@client_id = abc-abc-abc
@client_secret = 1q2w3e4r5t6y7u

// @name getTokenApi
POST {{base_url}}/oauth2/token
Content-Type: application/x-www-form-urlencoded; charset=utf-8
Accept: */*
Accept-Encoding: gzip, deflate

?client_id={{client_id}}
&client_secret={{client_secret}}

// get token from response
@access_token = {{getTokenApi.response.body.access_token}}
@refresh_token = {{getTokenApi.response.body.refresh_token}}
````
After executing getTokenApi in variables.http,
The @access_token variable will have a value. In another file (test-myapp.http), you can use this variable:
```
// file: test-myapp.http

{{ $import variables.http }}

@customer_id = 1234

// @name getCustomerInfo
GET {{base_url}}/v1/customers/{{customer_id}}
Authorization: Bearer {{access_token}}

```

I think this might help us.
